src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+    private static final String REPORT_HEADER = "Report for period ";
+    private static final String SEPARATOR = " - ";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public static String getSalaryInfo(
+            String[] names, String[] data, String dateFrom, String dateTo) {
+        LocalDate startDate = LocalDate.parse(dateFrom.trim(), DATE_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo.trim(), DATE_FORMATTER);
+
+        StringBuilder reportBuilder = new StringBuilder();
+        reportBuilder.append(REPORT_HEADER).append(dateFrom.trim())
+                .append(SEPARATOR).append(dateTo.trim()).append(LINE_SEPARATOR);
+
+        // Create Employee objects
+        Employee[] employees = new Employee[names.length];
+        for (int i = 0; i < names.length; i++) {
+            employees[i] = new Employee(names[i]);
+        }
+
+        // Iterate through the data array and create WorkRecord objects
+        for (String recordString : data) {
+            WorkRecord record = new WorkRecord(recordString);
+            LocalDate workDate = record.getDate();
+
+            // Check if the work date is within the specified range and update earnings
+            if (!workDate.isBefore(startDate) && !workDate.isAfter(endDate)) {
+                for (Employee employee : employees) {
+                    if (employee.getName().equals(record.getEmployeeName())) {
+                        employee.addEarnings(record.calculateEarnings());
+                    }
+                }
+            }
+        }
+
+        for (Employee employee : employees) {
+            reportBuilder.append(employee.getName()).append(" - ")
```

Comment:
Instead of concatenating strings with the '+' operator inside the loop, consider using the append method of StringBuilder for each part of the string to be appended. This will avoid creating unnecessary String objects and will be more efficient.